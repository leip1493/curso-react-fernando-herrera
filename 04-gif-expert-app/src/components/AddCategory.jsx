import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({ target: { value } }) => {
    setInputValue(value);    
  };

  const handleSubmit = e => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type="text" onChange={handleInputChange} value={inputValue} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};
