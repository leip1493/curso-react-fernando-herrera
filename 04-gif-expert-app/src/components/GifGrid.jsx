import React, { useEffect, useState } from "react";
import { GifGridItem } from "./GifGridItem";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category} </h3>
      {loading && <p className="animate__animated animate__flash">Loading</p>}
      <div className="card-grid">
        {gifs.map(gif => (
          <GifGridItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
