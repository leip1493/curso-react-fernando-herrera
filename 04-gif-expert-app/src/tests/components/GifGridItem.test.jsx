import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("<GifGridItem />", () => {
  const title = "Title";
  const url = "http://google.com/img.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Debe renderizar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un parrafo con el título", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("Debe tener la imágen y la url", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener la clase animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = "animate__fadeIn";

    expect(div.hasClass(className)).toBeTruthy(); // usando hasClass

    expect(div.prop("className").includes(className)).toBe(true)// usando prop("className").includes()
  });
});
