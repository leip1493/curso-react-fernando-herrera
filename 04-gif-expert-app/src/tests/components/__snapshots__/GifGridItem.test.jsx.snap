// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Debe renderizar correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="http://google.com/img.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
